### PR TITLE
Migrate to crate-universe

### LINKED ISSUE #374: Consider migrating from `cargo raze` to `crate_universe`
`cargo raze` used to provide more functionality than `rules_rust`'s own `crate_universe`, so we adopted it. My understanding is that `crate_universe` has seen substantial development since that initial decision, and we should now revisit it.

At the time of writing, we have the following open issues related to `cargo raze` itself, or our usage of it:
- https://github.com/vaticle/dependencies/issues/385
- https://github.com/vaticle/dependencies/issues/338
- https://github.com/vaticle/dependencies/issues/335
- https://github.com/vaticle/dependencies/issues/321

### PR DESCRIPTION
**Pull request by dmitrii-ubskii**
## What is the goal of this PR?

We upgrade rules_rust to v.0.16.1 and migrate from using cargo-raze to rules_rust's crate_universe for the purposes of translating cargo dependencies to bazel dependencies.

## What are the changes implemented in this PR?

Closes #374.

### CODE REVIEW

File: library/crates/update.sh [hunk 1]
```diff
@@ -25,8 +25,7 @@ CARGO_RAZE_HOME=$(cd "$(dirname "${path}")" && pwd -P)
 
 pushd "$CARGO_RAZE_HOME" > /dev/null
 
-cargo raze
-
-cp overrides/* remote/.
+cargo generate-lockfile
+CARGO_BAZEL_REPIN=true bazel sync --only=//library/rust/crates
```
Comment (dmitrii-ubskii) [inline library/crates/update.sh:29]: Update Cargo.lock and cargo-bazel-lock.json.

File: WORKSPACE [hunk 2]
```diff
@@ -88,8 +88,13 @@ maven(vaticle_dependencies_tool_maven_artifacts)
...
 
-load("//library/crates:crates.bzl", "raze_fetch_remote_crates")
-raze_fetch_remote_crates()
+load("@rules_rust//crate_universe:repositories.bzl", "crate_universe_dependencies")
+crate_universe_dependencies()
+
+load("//library/crates:crates.bzl", "fetch_crates")
+fetch_crates()
+load("@crates//:defs.bzl", "crate_repositories")
+crate_repositories()
```
Comment (dmitrii-ubskii) [inline WORKSPACE:97]: Sets up `crates`; needed here to be able to update cargo-bazel-lockfile.json. A similar construct replaces the raze call in the other projects' `WORKSPACE`s.

File: builder/rust/deps.bzl [hunk 3]
```diff
@@ -3,11 +3,8 @@ load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive", "http_file"
 def deps():
     http_archive(
         name = "rules_rust",
-        sha256 = "6bfe75125e74155955d8a9854a8811365e6c0f3d33ed700bc17f39e32522c822",
-        urls = [
-            "https://mirror.bazel.build/github.com/bazelbuild/rules_rust/releases/download/0.9.0/rules_rust-v0.9.0.tar.gz",
-            "https://github.com/bazelbuild/rules_rust/releases/download/0.9.0/rules_rust-v0.9.0.tar.gz",
-        ],
+        sha256 = "aaaa4b9591a5dad8d8907ae2dbe6e0eb49e6314946ce4c7149241648e56a1277",
+        urls = ["https://github.com/bazelbuild/rules_rust/releases/download/0.16.1/rules_rust-v0.16.1.tar.gz"],
```
Comment (dmitrii-ubskii) [inline builder/rust/deps.bzl:7]: Update rules-rust to 0.16.1 and, as a consequence, rust to 1.66.0 (from 1.62.1).

File: library/crates/Cargo.toml [hunk 4]
```diff
@@ -7,109 +7,33 @@ path = "" # ignored by cargo raze
 
 [dependencies]
 alcoholic_jwt = "=4091.0.0"
-async-trait = "0.1.59"
```
Comment (dmitrii-ubskii) [inline library/crates/Cargo.toml:10]: Removed unused crates, and crates pinned only for the benefit of raze metadata.

File: library/crates/Cargo.toml [hunk 5]
```diff
@@ -7,109 +25,33 @@ path = "" # ignored by cargo raze
...
+rocksdb = "=0.19.0"
+serde = "=1.0.152"
+serde_json = "=1.0"
+serial_test = "=0.9.0"
 sha2 = "=0.10.6"
-synthez = "=0.2.0"
-synthez-codegen = "=0.2.0"
-synthez-core = "=0.2.0"
-tokio = { version = "=1.20.1", features = ["rt", "rt-multi-thread"] }
+tokio = { version = "=1.24.1", features = ["rt", "rt-multi-thread"] }
```
Comment (dmitrii-ubskii) [inline library/crates/Cargo.toml:54]: Some crates' versions, such as chrono, were bumped as required by newly added crates and this bump to tokio.

**Review by alexjpwalker — APPROVED** [review]

**Pull request merged**